op-conductor/rpc/ws/flashblocks_handler.go

```diff
@@ -0,0 +1,257 @@
+	}
+
+	// Initialize the handler
+	handler := &Handler{
+		cfg:        cfg,
+		log:        log,
+		isLeaderFn: isLeaderFn,
+		metrics:    m,
+	}
+
+	// Try to establish initial connection to rollup boost WebSocket
+	maxConnectionAttempts := 5
+	var conn *websocket.Conn
+	var err error
+
+	for attempt := 1; attempt <= maxConnectionAttempts; attempt++ {
+		log.Info("attempting to connect to rollup boost WebSocket", "attempt", attempt, "url", cfg.RollupBoostWsURL)
+
+		// Create context with timeout for dialing
+		dialCtx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+		conn, _, err = websocket.Dial(dialCtx, cfg.RollupBoostWsURL, nil)
+		cancel()
+
+		if err == nil {
+			log.Info("successfully connected to rollup boost WebSocket")
+			handler.rollupBoostConn = conn
+			break
+		}
+
+		log.Warn("failed to connect to rollup boost WebSocket",
+			"attempt", attempt, "maxAttempts", maxConnectionAttempts, "err", err)
+
+		if attempt < maxConnectionAttempts {
+			time.Sleep(reconnectDelay)
+		}
+	}
+
+	// Still couldn't connect after all attempts
+	if err != nil {
+		log.Error("failed to connect to rollup boost WebSocket after multiple attempts",
+			"attempts", maxConnectionAttempts, "err", err)
+		return nil, fmt.Errorf("failed to connect to rollup boost WebSocket: %w", err)
+	}
+
+	return handler, nil
+}
+
+// Start initializes and starts the flashblocks handler
+func (h *Handler) Start(ctx context.Context) error {
+	// Start the WebSocket server
+	if err := h.startWebSocketServer(ctx); err != nil {
+		return err
+	}
+
+	// Start the rollup boost listener
+	h.rollupBoostCtx, h.rollupCancel = context.WithCancel(ctx)
+	go h.listenToRollupBoost(h.rollupBoostCtx)
+
+	return nil
+}
+
+// Stop closes any open WebSocket connections and shuts down the server
+func (h *Handler) Stop() {
+	// Signal the hub to stop if it exists
+	if h.hub != nil {
+		close(h.hub.done)
+	}
+
+	// Cancel the rollup boost context if it exists
+	if h.rollupCancel != nil {
+		h.rollupCancel()
+	}
+
+	// Close the rollup boost connection if it exists
+	if h.rollupBoostConn != nil {
+		h.log.Info("closing rollup boost WebSocket connection")
+		h.rollupBoostConn.Close(websocket.StatusNormalClosure, "shutting down")
+		h.rollupBoostConn = nil
+	}
+
+	// Close the HTTP server if it's running
+	if h.server != nil {
+		h.log.Info("closing WebSocket server")
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+
+		err := h.server.Shutdown(ctx)
+		if err != nil {
+			h.log.Error("Error shutting down WebSocket server", "err", err)
+		}
+		h.log.Info("WebSocket server closed")
+	}
+}
+
+// BroadcastMessage sends a message to all connected WebSocket clients
+func (h *Handler) BroadcastMessage(message []byte) {
+	h.hub.broadcast <- message
+}
+
+// Your startWebSocketServer should look like this:
+func (h *Handler) startWebSocketServer(_ context.Context) error {
```

Comment:
why not have a `WebSocketServer` struct that abstracts all this away? hub / serverMux, etc?

It could make start / stop easier, and also make testing more robust